src/export-plan/pdf-gen.ts

```diff
@@ -0,0 +1,328 @@
+        },
+        headStyles: {
+          fontSize: headerFontSize,
+          valign: 'middle',
+          halign: 'center',
+          fillColor: [255, 255, 255],
+          textColor: 0,
+          lineWidth: 0.5,
+          font: 'ProximaNova-Bold',
+          lineColor: [216, 216, 216],
+        },
+        columnStyles: {
+          0: { cellWidth: 266 },
+          1: { cellWidth: 50 },
+          2: { cellWidth: 200 },
+        },
+        didParseCell: data => {
+          data.cell.styles.lineColor = [216, 216, 216];
+          data.cell.styles.lineWidth = 0.5;
+        },
+        willDrawCell: data => {
+          doc.setFillColor(255, 255, 255);
+          if (data.row.index === sem.courses.length - 1) {
+            doc.setDrawColor(255, 255, 255);
+            data.cell.styles.lineWidth = 0.5;
+          }
+        },
+        /* eslint no-loop-func: "off" */
+        didDrawCell: data => {
+          if (data.row.index === sem.courses.length - 1) {
+            doc.setDrawColor(216, 216, 216);
+            doc.setLineWidth(0.5);
+            // draw middle vertical borders for last row
+            if (data.column.index > 0)
+              doc.line(data.cell.x, data.cell.y + data.row.height, data.cell.x, data.cell.y);
+            // draw top horizontal border for last row
+            doc.line(data.cell.x + data.column.width, data.cell.y, data.cell.x, data.cell.y);
+          }
+
+          tableHeight = data.cell.y + data.row.height - startct + rowHeight;
+
+          if (
+            data.column.index === 2 &&
+            data.section === 'body' &&
+            data.row.index >= 0 &&
+            data.row.index < body.length
+          ) {
+            let yPos = data.cell.y + 3;
+            groups[data.row.index].forEach((group, index) => {
+              const xPos =
+                data.cell.x + doc.getTextWidth(body[data.row.index][2].split('\n')[index]) + 8;
+
+              const colour = colours[data.row.index][index];
+              renderBubbles(doc, xPos, yPos, group, colour);
+              yPos += rowFontSize + 1.5;
+            });
+          }
+        },
+      });
+    }
+
+    doc.setDrawColor(216, 216, 216);
+    doc.setLineWidth(0.5);
+    doc.roundedRect(tableX, startct - rowHeight, tableWidth, tableHeight, 4, 4);
+    startct += tableHeight + tableGap;
+  }
+
+  const footerY = doc.internal.pageSize.height - 25;
+  const footerX =
+    (doc.internal.pageSize.width - doc.getTextWidth('downloaded from courseplan.io')) / 2;
+  doc.text('downloaded from ', footerX, footerY, { align: 'left' });
+  doc.textWithLink('courseplan.io', footerX + doc.getTextWidth('downloaded from '), footerY, {
+    url: 'https://courseplan.io',
+  });
+  doc.setDrawColor(0);
+  doc.line(
+    footerX + doc.getTextWidth('downloaded from '),
+    footerY + 3,
+    footerX + doc.getTextWidth('downloaded from courseplan.io'),
+    footerY + 3
+  );
+  const pdfName = `${store.state.userName.firstName}'s CoursePlan`;
```

Comment:
Can we remove the `'s` in the name? I think special characters might make the file name a little weird. For example, the file you link in your PR downloads as `Vaishnavi.s.CoursePlan-12`.

Reply:
Huh, this is weird. What browser are you using? I tried with both Chrome and Safari and the quotation works for me. The link in my PR is just a file that I uploaded, so I think that might be on github's end.

Also, are you still getting that addImage error? i am not getting that on either chrome or safari.

Reply:
see my recent comment. I haven't actually been able to download it yet, so I'm not sure how it will be stored. However, I think using special characters in file names generally can be problematic.

Reply:
also the spaces can be annoying because they introduce `%20`

Reply:
Hm, where do they introduce %20? I guess I can double check with the design folks about what file name they'd prefer. But maybe since this is meant for the end user it, I thought it'd make sense to capitalize it etc. Otherwise, I can just do `vaishnavi-course-plan.pdf`.